src/Templates/SingleProductTemplateCompatibility.php

```diff
@@ -0,0 +1,413 @@
+					'before'
+				),
+				$block_content,
+				$this->get_hooks_buffer(
+					$first_block_hook['after'],
+					'after'
+				)
+			);
+		}
+
+		if ( isset( $block['attrs'][ self::IS_LAST_BLOCK ] ) ) {
+			return sprintf(
+				'%1$s%2$s%3$s',
+				$this->get_hooks_buffer( array(), 'before' ),
+				$block_content,
+				$this->get_hooks_buffer(
+					$last_block_hook['after'],
+					'after'
+				)
+			);
+		}
+	}
+
+	/**
+	 * Update the render block data to inject our custom attribute needed to
+	 * determine which is the first block of the Single Product Template.
+	 *
+	 * @param array         $parsed_block The block being rendered.
+	 * @param array         $source_block An un-modified copy of $parsed_block, as it appeared in the source content.
+	 * @param WP_Block|null $parent_block If this is a nested block, a reference to the parent block.
+	 *
+	 * @return array
+	 */
+	public function update_render_block_data( $parsed_block, $source_block, $parent_block ) {
+		return $parsed_block;
+	}
+
+	/**
+	 * Set supported hooks.
+	 */
+	protected function set_hook_data() {
+		$this->hook_data = array(
+			'woocommerce_before_single_product'         => array(
+				'block_name' => '',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_output_all_notices' => 10,
+					'woocommerce_breadcrumb'         => 20,
+				),
+			),
+			'woocommerce_before_single_product_summary' => array(
+				'block_name' => '',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_show_product_sale_flash' => 10,
+					'woocommerce_show_product_images'     => 20,
+				),
+			),
+			'woocommerce_single_product_summary'        => array(
+				'block_name' => '',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_template_single_title'   => 5,
+					'woocommerce_template_single_rating'  => 10,
+					'woocommerce_template_single_price'   => 10,
+					'woocommerce_template_single_excerpt' => 20,
+					'woocommerce_template_single_add_to_cart' => 30,
+					'woocommerce_template_single_meta'    => 40,
+					'woocommerce_template_single_sharing' => 50,
+				),
+			),
+			'woocommerce_after_single_product'          => array(
+				'block_name' => '',
+				'position'   => 'after',
+				'hooked'     => array(),
+			),
+			'woocommerce_share'                         => array(
+				'block_name' => 'woocommerce/product-details',
+				'position'   => 'before',
+				'hooked'     => array(),
+			),
+			'woocommerce_after_single_product_summary'  => array(
+				'block_name' => 'woocommerce/product-details',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_output_product_data_tabs' => 10,
+					'woocommerce_upsell_display'           => 15,
+					'woocommerce_output_related_products'  => 20,
+				),
+			),
+		);
+	}
+
+
+	/**
+	 * Check if the current template is a single product template.
+	 *
+	 * @return bool
+	 */
+	private function is_single_product_template() {
+		return is_product();
```

Comment:
> I'd recommend ditching this is_single_product_template() method and relying directly on is_product().

👍 

> Additionally, this function is unaware of any templates; it will return true whenever you visualize a single post with the type = product.

Right, but according to WP hierarchy, if you are visualizing a single post with type product, you are visualizing the Single Product Template, no?

Reply:
> Right, but according to WP hierarchy, if you are visualizing a single post with type product, you are visualizing the Single Product Template, no?

Correct, but just as a coincidence, not because `is_product()` is verifying the template's name. A single post on WP can have multiple templates (as we are already in the process of implementing for each one of the product types).

Reply:
> WC core uses this function to check if it is loaded a Single Product page ([source code](https://github.com/woocommerce/woocommerce/blob/3681b84a4c524691ad0748d29992a2d75df87fde/plugins/woocommerce/includes/class-wc-frontend-scripts.php/#L402-L416)).

Yep! My only observation here is regarding the name of the method in particular: as `is_single_product_template()` was "wrapping/renaming" `is_product()`,  and that's not a valid assumption because `is_product` is only checking for the existence of a single post of a type = 'product'. Thanks for addressing the feedback!